R/check_target_file_ext_valid.R

```diff
@@ -0,0 +1,42 @@
+#' Check that a target data file has a valid extension.
+#'
+#' Note that files which are part of a hive partitioned dataset must have
+#' parquet file extension only.
+#' @inheritParams check_target_file_name
+#' @inherit check_valid_round_id return
+#' @export
+check_target_file_ext_valid <- function(file_path) {
+  hive <- is_hive_partitioned_path(file_path, strict = FALSE)
+  if (hive) {
+    valid_ext <- c("parquet")
+  } else {
+    valid_ext <- c("csv", "parquet")
+  }
+  file_ext <- fs::path_ext(file_path)
+
+  check <- file_ext %in% valid_ext
+  details <- NULL
+
+  if (!check) {
+    invalid_ext <- setdiff(file_ext, valid_ext) # nolint: object_usage_linter
+    details <- cli::format_inline(
+      "Extension {.val {invalid_ext}} is not.
+    Must be one of {.val {valid_ext}}."
+    )
+  }
+
+  subject <- if (hive) {
+    "Hive-partitioned target data file"
+  } else {
+    "Target data file"
+  }
```

Comment:
Mainly to simplify. Feel free to ignore

```suggestion
  if (hive) {
    valid_ext <- c("parquet")
    subject <- "Hive-partitioned target data file"
  } else {
    valid_ext <- c("csv", "parquet")
    subject <- "Target data file"
  }
  file_ext <- fs::path_ext(file_path)

  check <- file_ext %in% valid_ext
  details <- NULL

  if (!check) {
    invalid_ext <- setdiff(file_ext, valid_ext) # nolint: object_usage_linter
    details <- cli::format_inline(
      "Extension {.val {invalid_ext}} is not.
    Must be one of {.val {valid_ext}}."
    )
  }
```

Reply:
Nice!